ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1641,9 +1645,9 @@ leaderId, getMemberId(), currentTerm, followerCommit, inconsistencyReplyNextInde
       state.updateConfiguration(entries);
     }
     future.join();
+    final CompletableFuture<Void> appendLog = entries.isEmpty()? appendLogFuture.get()
```

Comment:
@SzyWilliam , I see the problem now -- when it is empty, it should not wait for the previous append.  Otherwise, it blocks the heartbeats.
```java
    final CompletableFuture<Void> appendLog = entries.isEmpty()? CompletableFuture.completedFuture(null)
```

Reply:
That is a tricky one. Thanks for spotting and troubleshooting this!